### ISSUE TITLE
Prisma client fails build

### ISSUE THREAD

**Issue by jeremygottfried** [dealmore/terraform-aws-next-js#73]
Build fails for projects using prisma client.

```sh
Type error: Module '"../../../node_modules/@prisma/client"' has no exported member 'MyModel'.
```

> `import { MyModel } from '@prisma/client';` 

> Build failed:
NowBuildError: Command "yarn run build" exited with 1
    at ChildProcess.<anonymous> (/Users/jeremygottfried/Development/sweepstakes-app/sweepstakes-next-app/node_modules/@vercel/build-utils/dist/index.js:29311:20)
    at ChildProcess.emit (events.js:310:20)
    at ChildProcess.EventEmitter.emit (domain.js:482:12)
    at maybeClose (internal/child_process.js:1021:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5) {
  hideStackTrace: true,
  code: 'BUILD_UTILS_SPAWN_1',
  link: undefined,
  action: undefined
}

**Comment by ofhouse** [comment]
Can you share more information about your project?

Seems more like a TypeScript error in the first place.
I tried out the [official JavaScript example](https://github.com/prisma/prisma-examples/tree/latest/javascript/rest-nextjs) and the build looks fine to me.

However using the TypeScript example gives me a lot of TypeScript related errors.
Have you tried deploying it to Vercel, does it work there (Asking because we use a similar build process as Vercel under the hood)?

**Comment by jeremygottfried** [comment]
Yes it's a typescript error.
> Can you share more information about your project?
> 
> Seems more like a TypeScript error in the first place.
> I tried out the [official JavaScript example](https://github.com/prisma/prisma-examples/tree/latest/javascript/rest-nextjs) and the build looks fine to me.
> 
> However using the TypeScript example gives me a lot of TypeScript related errors.

Yeah I have deployed it to Vercel. I don't seem to have any typescript related issues there.
> Have you tried deploying it to Vercel, does it work there (Asking because we use a similar build process as Vercel under the hood)?

Here's my tsconfig
```json
{
  "compilerOptions": {
    "sourceMap": true,
    "outDir": "dist",
    "strict": true,
    "lib": [
      "esnext"
    ],
    "esModuleInterop": true,
    "target": "es5",
    "allowJs": true,
    "skipLibCheck": true,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve"
  },
  "include": [
    "next-env.d.ts",
    "**/*.ts",
    "**/*.tsx"
  ],
  "exclude": [
    "node_modules"
  ]
}
```

tf-next version: 0.5.0
ts-node version: 9.1.1
typescript version: 4.1.3
prisma/cli version: 2.18.0
prisma/client version: 2.16.1
next version: 10.0.5

Here's next.config.js:
```js
const withTM = require('next-transpile-modules')(['react-spring']); // pass the modules you would like to see transpiled
const path = require("path");

module.exports = withTM();
```

main.tf:

```tf
# main.tf

terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 3.0"
    }
  }
}

# Main region where the resources should be created in
provider "aws" {
  region = "us-east-1"
}

module "tf_next" {
  source = "dealmore/next-js/aws"
}

output "cloudfront_domain_name" {
  value = module.tf_next.cloudfront_domain_name
}
```

**Comment by ofhouse** [comment]
Sorry for the radio silence here.
Saw a similar error with Blitz.js and Prisma today and it seems that `tf-next` has some problems with prisma when it runs it inside the temporary directory it creates for every build.

As a workaround you can run `tf-next build --skipDownload` which runs `tf-next` in the same directory instead of creating a temporary one.
This fixed the issue for me (note that `tf-next` makes some changes to your `next.config.js` then that should not commited) but will investigate further why the build fails in the temporary directory.

**Comment by jeremygottfried** [comment]
Great thanks I'll try the workaround when I have a chance!

**Comment by ofhouse** [comment]
Okay, finally found the main issue.
Prisma uses the `NOW_BUILDER` environment variable to detect the build environment in Vercel builds.
https://github.com/prisma/prisma/blob/1d0045fe60dc1992173f3f5be84b24129f0d45a3/src/packages/cli/scripts/install.js#L10

Since our build environment is forked from the Vercel environment we also need to set this environment variable in order to get the workaround from Prisma correctly applied.
It could work to set the `NOW_BUILDER=#1'` environment variable manually before running `tf-next` (not checked) but I prepare a general fix for this issue in #85.

**Comment by ofhouse** [comment]
Just released [`tf-next@0.6.0`](https://www.npmjs.com/package/tf-next) which should fix the issue.
Upgrading `tf-next` in your `package.json` should be enough to make it work, no update of the Terraform module required.

**Comment by ofhouse** [comment]
Ok there was still a small error, which caused the types generation to take place in the wrong directory.
This was fixed in #87 and released in [`tf-next@0.6.1`](https://www.npmjs.com/package/tf-next).
Closing this for now.
